waku/factory/node_factory.nim

```diff
@@ -113,6 +113,12 @@ proc initNode(
 
 ## Mount protocols
 
+proc getNetworkShards*(conf: WakuNodeConf): uint32 =
+  if conf.networkShards != 0:
+    return conf.networkShards
+  # If conf.networkShards is not set, use the number of shards configured as networkShards
```

Comment:
Maybe we should also comment that in the `external_config.nim` parameter so that it is clear that the `networkShards` is optional.